pkg/apiserver/api_installer.go

```diff
@@ -948,3 +973,19 @@ var _ rest.StorageMetadata = defaultStorageMetadata{}
 func (defaultStorageMetadata) ProducesMIMETypes(verb string) []string {
 	return nil
 }
+
+// isSubresource checks if the given storage path is the path of a subresource and returns
+// the resource and subresource components.
+func isSubresource(path string) (string, string, bool, error) {
+	var resource, subresource string
+	switch parts := strings.Split(path, "/"); len(parts) {
+	case 2:
+		resource, subresource = parts[0], parts[1]
+	case 1:
+		resource = parts[0]
+	default:
+		// TODO: support deeper paths
+		return "", "", false, fmt.Errorf("api_installer allows only one or two segment paths (resource or resource/subresource)")
+	}
+	return resource, subresource, len(subresource) > 0, nil
```

Comment:
This bool doesn't seem to add much value.  Why not return a single field that can't possibly disagree with itself?